### PR TITLE
client: provide pointers requests for signing

### PR DESCRIPTION
**Pull request by fyrchik**
Signed-off-by: Evgenii Stratonikov <evgeniy@nspcc.ru>

### CODE REVIEW

**Review by carpawell — APPROVED** [review]

**Review by alexvanin — APPROVED** [review]

**Review by KirillovDenis — DISMISSED** [review]
This fix panic `unsupported session message object.HeadRequest`. Checked with rest-gw and aio v0.29.0/v0.30.0

**Comment by fyrchik** [comment]
Also fixed signing of `GetRange` request

File: client/object_get.go [hunk 1]
```diff
@@ -686,6 +686,16 @@ func (c *Client) ObjectRangeInit(ctx context.Context, prm PrmObjectRange) (*Obje
 	req.SetBody(&body)
 	c.prepareRequest(&req, &prm.meta)
 
+	key := prm.key
+	if key == nil {
+		key = &c.prm.key
+	}
+
+	err := signature.SignServiceMessage(key, &req)
```
Comment (carpawell) [inline client/object_get.go:694]: move that before context init to satisfy the linter?

Comment (carpawell) [reply client/object_get.go:694]: outdated

**Review by carpawell — APPROVED** [review]

**Review by KirillovDenis — APPROVED** [review]

**Pull request merged**